packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/GoogleMapMarkerController.m

```diff
@@ -240,17 +240,44 @@ - (void)interpretMarkerOptions:(NSDictionary *)data
         } else if([markerType isEqualToString:@"rounded"]) {
             NSString *label = data[@"label"];
             if(label && label != (id)[NSNull null]) {
-                UIImage *img = [self roundedMarkerImageWithText:label];
-                [self setIcon:img];
+                UIImage *image = [self roundedMarkerImageWithText:label withMarkerColor:UIColor.whiteColor withTextColor:UIColor.blackColor];
+                [self setIcon:image];
             } else {
                 NSString *error =
                 [NSString stringWithFormat:@"label was not provided."];
                 NSException *exception = [NSException exceptionWithName:@"InvalidBitmapDescriptor"
                                                                  reason:error
                                                                userInfo:nil];
                 @throw exception;
-            }           
+            }
+         }else if([markerType isEqualToString:@"rounded_visited"]) {
+            NSString *label = data[@"label"];
+            if(label && label != (id)[NSNull null]) {
+                UIImage *image = [self roundedMarkerImageWithText:label withMarkerColor:UIColor.whiteColor withTextColor:[UIColor colorWithRed:(110.0f/255.0f) green:(110.0f/255.0f) blue:(100.0f/255.0f) alpha:1]];
```

Comment:
Since the visited and selected markers are the same shape i added two parameters to change the text and marker color